### PR TITLE
Fix #1223 FindOrCreate missing error callback

### LINKED ISSUE #1223: FindOrCreate does not return errors on connector error
### Bug or feature request
- [x] Bug

### Description of feature (or steps to reproduce if bug)
There is a bug in dao.js that does not produce an error when it should on a connector issue. This yields issues for mongodb in particular. 

Reference as to fix:
https://github.com/strongloop/loopback-connector-mongodb/issues/296

Errors are overwritten by Model.notifyObserversOf should insert conditional and return just after findOrCreate callback (please :)

```
function _findOrCreate(query, data, currentInstance) {
    var modelName = self.modelName;
    function findOrCreateCallback(err, data, created) {
      console.log(!!err); //INSERT HERE: if (err) { return cb(err); }
      var context = {
        Model: Model,
        data: data,
        isNewInstance: created,
        hookState: hookState,
        options: options,
      };
      Model.notifyObserversOf('loaded', context, function(err) {
        if (err) return cb(err);

        var obj, Model = self.lookupModel(data);
```

### Additional information (Node.js version, LoopBack version, etc)
This Error is present in:
- 2.53.0 (one I'm using)
But also in your 3.0.0

### PR DESCRIPTION
**Pull request by jamesjjk**
Added missing error callback to findOrCreateCallback

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by ssh24** [comment]
@jamesjjk Thank you for the contribution. Could you please sign the [Contributors License Agreement](https://cla.strongloop.com/agreements/strongloop/loopback-datasource-juggler)? Also please rebase your branch.

**Comment by jamesjjk** [comment]
@ssh24 Since the commit has been made is there any reason to re-integrate at this stage?

**Comment by ssh24** [comment]
@jamesjjk Your branch is not updated with the master of `loopback-datasource-juggler`. That is why you need to rebase the branch.

You could also give me access to your fork by clicking `Allow maintainers to edit` on the bottom right of the page.

File: lib/dao.js [hunk 1]
```diff
@@ -1037,6 +1037,7 @@ DataAccessObject.findOrCreate = function findOrCreate(query, data, options, cb)
   function _findOrCreate(query, data, currentInstance) {
     var modelName = self.modelName;
     function findOrCreateCallback(err, data, created) {
+      if(err) { return cb(err); }
```
Comment (loay) [inline lib/dao.js]: Just use `if (err) return cb(err)`

**Comment by dhmlau** [comment]
@jamesjjk , could you please sign the CLA from https://cla.strongloop.com/agreements/strongloop/loopback-datasource-juggler?

Also, do you need help to rebase?  
> You could also give me access to your fork by clicking Allow maintainers to edit on the bottom right of the page.

**Comment by dhmlau** [comment]
@jamesjjk, please sign the CLA form by this Friday (Jul 14), otherwise I'm afraid we'll have to close your PR.  Thanks!

**Comment by dhmlau** [comment]
@slnode  test please

**Comment by ssh24** [comment]
This issue is fixed. See https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/dao.js#L1100

**Pull request closed without merging**